hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableAsSelectCommand.scala

```diff
@@ -41,6 +41,10 @@ case class CreateHoodieTableAsSelectCommand(
    mode: SaveMode,
    query: LogicalPlan) extends DataWritingCommand {
 
+  def withNewChildInternal(newChild: LogicalPlan): CreateHoodieTableAsSelectCommand = {
```

Comment:
would you please clarify why CreateHoodieTableAsSelectCommand can not extends HoodieLeafRunnableCommand?

Reply:
`CreateHoodieTableAsSelectCommand ` has already extended `DataWritingCommand` and then `UnaryLike` which overrides `withNewChildrenInternal` with `final` keyword and declare `withNewChildInternal ` for subclasses.

`trait DataWritingCommand extends UnaryCommand`
`trait UnaryCommand extends Command with UnaryLike[LogicalPlan]`

![image](https://user-images.githubusercontent.com/10036681/145708193-9acd04ef-1e79-4cd8-af08-d55e9e874026.png)